civis/tests/test_resources.py

```diff
@@ -42,20 +42,6 @@
     The last modification time for this credential.""")  # noqa: E122
 
 
-def test_create_method_iterator_kwarg():
-    args = [{"name": 'limit', "in": 'query', "required": False, "doc": ""},
-            {"name": 'page_num', "in": 'query', "required": False, "doc": ""},
-            {"name": 'order', "in": 'query', "required": False, "doc": ""},
-            {"name": 'order_by', "in": 'query', "required": False, "doc": ""}]
-    method = _resources.create_method(args, 'get', 'mock_name', '/objects',
-                                      'fake_doc')
-    mock_endpoint = mock.MagicMock()
-
-    method(mock_endpoint, iterator=True)
-    mock_endpoint._call_api.assert_called_once_with(
-        'get', '/objects', {}, {}, iterator=True)
```

Comment:
I think this test is actually okay.  `iterator` is in the spec for `_call_api`, since `_call_api` accepts `**kwargs` (not sure if this is actually for the best, though).   [Here](https://github.com/civisanalytics/civis-python/blob/5c135f9482ed6ca00c83871fb40b3cd35fc3045c/civis/base.py#L109) is the corresponding line.

However, this test should fail if `{"name": 'limit', ...}` and `{"name": 'page_num'}` isn't included in `args`, since this would cause [`iterable_method`](https://github.com/civisanalytics/civis-python/blob/master/civis/resources/_resources.py#L227) to return`False`, and then [`raise_for_unexpected_kwargs`](https://github.com/civisanalytics/civis-python/blob/master/civis/resources/_resources.py#L250) should raise a `TypeError`.

This is pretty confusing, does that trail of breadcrumbs follow your understanding as well?

Reply:
Ah, I see. I'd gotten confused about what the `iterator` argument was doing. I thought it was part of the Platform spec, but it's a feature of the API client. I needed to modify `create_method` to pop out `iterator` before binding the arguments, but the test is restored. This means that `iterator` is a keyword-only argument in both Python 2 and Python 3, and is the only keyword-only argument in Python 2.